backend/tracim_backend/models/data.py

```diff
@@ -1139,19 +1142,26 @@ def owner(self, value: User) -> None:
     def owner(cls) -> InstrumentedAttribute:
         return ContentRevisionRO.owner
 
-    @hybrid_property
+    @property
     def children(self) -> ['Content']:
         """
         :return: list of children Content
         :rtype Content
         """
-        # Return a list of unique revisions parent content
-        # TODO - G.M - 2018-10-24 - Set is just here to avoid distinct object,
-        # check if possible to do this better, probably using database sorting
-        # and sql distinct.
-        revisions = set([revision.node for revision in self.children_revisions])
-        revisions = sorted(revisions, key=lambda r: r.revision_id)
-        return revisions
+        # TODO - G.M - 2019-02-08 - Refactor this code to be more efficient
+        all_childrens_revisions = object_session(self).query(ContentRevisionRO).filter( ContentRevisionRO.parent_id == self.content_id).all()
+        all_up_to_date_revisions_result = object_session(self).query(
+            func.max(ContentRevisionRO.revision_id)
+            ).group_by(ContentRevisionRO.content_id).all()
```

Comment:
you take the full database, here. What about at least filtering by content_id found in all_children_revisions ?

Here is the code

```
all_childrens_revisions = object_sessio  # unchanged
children_content_ids = set([cr.content_id for cr in all_childrens_revisions])
all_up_to_date_revisions_result = object_session(self).query(
    func.max(ContentRevisionRO.revision_id)
    ) \
    .filter(ContentRevisionRO.content_id.in_(children_content_ids) \
    .group_by(ContentRevisionRO.content_id).all()

Reply:
nice idea. object_session is a method to retrieve session context from sqlalchemy object, so it's really all database. Thats really different from using query object from libContent where there are filter.

Reply:
done.